haystack/core/super_component/super_component.py

```diff
@@ -0,0 +1,354 @@
+
+        :returns:
+            Dictionary with serialized data.
+        """
+        return self._to_super_component_dict()
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "SuperComponent":
+        """
+        Deserializes the SuperComponent from a dictionary.
+
+        Must be overwritten for custom component implementations that inherit from SuperComponent.
+
+        :param data: The dictionary to deserialize from.
+        :returns:
+            The deserialized SuperComponent.
+        """
+        pipeline = Pipeline.from_dict(data["init_parameters"]["pipeline"])
+        data["init_parameters"]["pipeline"] = pipeline
+        return default_from_dict(cls, data)
+
+    def run(self, **kwargs: Any) -> Dict[str, Any]:
+        """
+        Runs the wrapped pipeline with the provided inputs.
+
+        Steps:
+        1. Maps the inputs from kwargs to pipeline component inputs
+        2. Runs the pipeline
+        3. Maps the pipeline outputs to the SuperComponent's outputs
+
+        :param kwargs: Keyword arguments matching the SuperComponent's input names
+        :returns:
+            Dictionary containing the SuperComponent's output values
+        """
+        filtered_inputs = {param: value for param, value in kwargs.items() if value != _delegate_default}
+        pipeline_inputs = self._map_explicit_inputs(input_mapping=self.input_mapping, inputs=filtered_inputs)
+        pipeline_outputs = self.pipeline.run(data=pipeline_inputs)
+        return self._map_explicit_outputs(pipeline_outputs, self.output_mapping)
+
+    @staticmethod
+    def _split_component_path(path: str) -> Tuple[str, str]:
+        """
+        Splits a component path into a component name and a socket name.
+
+        :param path: A string in the format "component_name.socket_name".
+        :returns:
+            A tuple containing (component_name, socket_name).
+        :raises InvalidMappingError:
+            If the path format is incorrect.
+        """
+        comp_name, socket_name = parse_connect_string(path)
+        if socket_name is None:
+            raise InvalidMappingError(f"Invalid path format: '{path}'. Expected 'component_name.socket_name'.")
+        return comp_name, socket_name
+
+    def _validate_input_mapping(
+        self, pipeline_inputs: Dict[str, Dict[str, Any]], input_mapping: Dict[str, List[str]]
+    ) -> None:
+        """
+        Validates the input mapping to ensure that specified components and sockets exist in the pipeline.
+
+        :param pipeline_inputs: A dictionary containing pipeline input specifications.
+        :param input_mapping: A dictionary mapping wrapper input names to pipeline socket paths.
+        :raises InvalidMappingError:
+            If the input mapping is invalid or contains nonexistent components or sockets.
+        """
+        if not isinstance(input_mapping, dict):
+            raise InvalidMappingError("input_mapping must be a dictionary")
```

Comment:
nit: maybe here a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError) makes more sense

Reply:
Good point to discuss. I think when users see `InvalidMappingError` they immediately understand what kind of issue they are facing and where to fix it. In contrast, a generic `TypeError` (or in some other places `ValueError`) could mean anything. With the `InvalidMappingError` we can nicely group all kinds of errors that come from the mapping. Let's get another quick opinion from @sjrl who came up with this I believe.

Reply:
An alternative could be to add a `InvalidMappingTypeError(TypeError)` and a `InvalidMappingValueError(ValueError)` and then use them accordingly instead of `InvalidMappingError`.
For example, we could raise an `InvalidMappingValueError` in `_split_component_path`.